module/VuFind/src/VuFind/ILS/Driver/MultiBackend.php

```diff
@@ -1235,21 +1242,23 @@ public function cancelILLRequests($cancelDetails)
      * submitted as form data. This value is then extracted by the CancelILLRequests
      * function.
      *
-     * @param array $details An array of item data
+     * @param array $request An array of request data
+     * @param array $patron  The patron array from patronLogin
      *
      * @return string Data for use in a form field
      */
-    public function getCancelILLRequestDetails($details)
+    public function getCancelILLRequestDetails($request, $patron)
     {
-        $source = $this->getSource($details['id'] ?? $details['item_id'] ?? '');
+        $source = $this->getSource($patron['cat_username']);
         $driver = $this->getDriver($source);
         if ($driver
             && $this->methodSupported(
-                $driver, 'getCancelILLRequestDetails', compact('details')
+                $driver, 'getCancelILLRequestDetails', compact('request', 'patron')
```

Comment:
I'm reasonably sure that methodSupported() doesn't care that the first key of the compacted array has changed from 'details' to 'request' but just double-checking since it's a subtle behavior change.

Reply:
It doesn't care, but I think is actually completely borked and should not be using compact at all! The parameter is passed as the second argument to supportsMethod, which returns "True if the method can be called with the given parameters, false otherwise", so I believe this should be `[$request, $patron]` instead.

Reply:
Yes, that's my impression too. We can leave it alone here since it's really a separate concern, but maybe we need a separate PR to eliminate the unnecessary compacts in the MultiBackend driver. Would you like me to put that together?

Reply:
I think it's my bad, so I'll volunteer to clean it up. :)

Reply:
Thanks! As we discussed on Slack, that can be addressed in a separate PR since the compact problem is largely independent of what's being worked on here.